anesthetic/plot.py

```diff
@@ -59,6 +102,135 @@ def _constructor(self):
     def _constructor_sliced(self):
         return AxesSeries
 
+    def get_axes(self, labels=None, lower=True, diagonal=True, upper=True,
+                 ticks='outer', gridspec_kw=None, subplot_spec=None, **fig_kw):
+        """Create a set of axes for plotting 2D marginalised posteriors."""
+        all_params = list(self.columns) + list(self.index)
+        if labels is None:
+            labels = {}
+        labels = {p: labels[p] if p in labels else p for p in all_params}
+        if gridspec_kw is None:
+            gridspec_kw = {}
+
+        for j, y in enumerate(self.index):
+            for i, x in enumerate(self.columns):
+                if all_params.index(x) < all_params.index(y):
+                    if lower:
+                        self[x][y] = -1
+                elif all_params.index(x) > all_params.index(y):
+                    if upper:
+                        self[x][y] = +1
+                elif diagonal:
+                    self[x][y] = 0
+
+        self.dropna(axis=0, how='all', inplace=True)
+        self.dropna(axis=1, how='all', inplace=True)
+
+        fig = fig_kw.pop('fig') if 'fig' in fig_kw else plt.figure(**fig_kw)
+        if self.shape[0] != 0 and self.shape[1] != 0:
+            hspace = gridspec_kw.pop('hspace', 0)
+            wspace = gridspec_kw.pop('wspace', 0)
+            if subplot_spec is None:
+                gs = GridSpec(*self.shape, hspace=hspace, wspace=wspace,
+                              **gridspec_kw)
+            else:
+                gs = GridSpecFromSubplotSpec(*self.shape,
+                                             hspace=hspace, wspace=wspace,
+                                             subplot_spec=subplot_spec,
+                                             **gridspec_kw)
+
+        if self.size == 0:
+            return fig, self
+        position = self.copy()
+        self[:][:] = None
+        for j, y in enumerate(self.index[::-1]):
+            for i, x in enumerate(self.columns):
+                if position[x][y] is not None:
+                    sx = list(self[x].dropna())
+                    sx = sx[0] if sx else None
+                    sy = list(self.T[y].dropna())
+                    sy = sy[0] if sy else None
+                    self[x][y] = fig.add_subplot(gs[self.index.size-1-j, i],
+                                                 sharex=sx, sharey=sy)
+
+                    if position[x][y] == 0:
+                        self[x][y].twin = self[x][y].twinx()
+                        self[x][y].twin.set_yticks([])
+                        self[x][y].twin.set_ylim(0, 1.1)
+                        make_diagonal(self[x][y])
+                        self[x][y].position = 'diagonal'
+                        self[x][y].twin.xaxis.set_major_locator(
+                            MaxNLocator(3, prune='both'))
```

Comment:
I think we've resigned ourselves to having a special paired axis on the diagonal, but this is clunky way of doing it -- is there  way to have a `DiagonalAxes` class which is defined in the body of this file, and created directly, rather than adjusted after the fact?

Reply:
I tried using `matplotlib.axes._subplots.SubplotBase` as a parent class, but that did not work. The parent class of `DiagonalAxes` is not a specific matplotlib base class, but somehow is connected to the specific GridSpec instance...

I'll turn `make_diagonal` into a `@staticmethod`, though.